### PR TITLE
Allow string parameters for criterion in torch param

### PR DESCRIPTION
**Pull request by marxprop**
This PR ensures the torch dict parameter is totally serializable. It allows the torch_dict params to accept the loss value as a string. It initializes the criterion by fetching the class from `torch.nn` module. It also allows the use of custom defined loss in dgbtorch.py.

Acceptable strings:
- Loss class names defined in dgbtorch.py (e.g DiceLoss)
- Loss class names defined in torch.nn (e.g CrossEntropyLoss) - users are expected to use the pytorch class name as the parameter
- None, this initializes the right loss class by default

### CODE REVIEW

**Comment by marxprop** [comment]
@arnauddgbes , please disregard the failed test, the PR also comes with the proper changes for the tests

**Review by arnauddgbes — APPROVED** [review]

**Review by arnauddgbes — APPROVED** [review]

**Review by arnauddgbes — APPROVED** [review]
